api/src/main/java/openconsensus/trace/BigendianEncoding.java

```diff
@@ -49,43 +51,49 @@
 
   /**
    * Returns the {@code long} value whose big-endian representation is stored in the first 8 bytes
-   * of {@code bytes} starting from the {@code offset}.
+   * of {@code bytes}.
    *
-   * @param bytes the byte array representation of the {@code long}.
-   * @param offset the starting offset in the byte array.
+   * <p>The returned value will be read as big-endian independently of the value of {@code
+   * byte.order()}.
+   *
+   * @param bytes the byte buffer representation of the {@code long}.
    * @return the {@code long} value whose big-endian representation is given.
    * @throws IllegalArgumentException if {@code bytes} has fewer than 8 elements.
    */
-  static long longFromByteArray(byte[] bytes, int offset) {
-    Utils.checkArgument(bytes.length >= offset + LONG_BYTES, "array too small");
-    return (bytes[offset] & 0xFFL) << 56
-        | (bytes[offset + 1] & 0xFFL) << 48
-        | (bytes[offset + 2] & 0xFFL) << 40
-        | (bytes[offset + 3] & 0xFFL) << 32
-        | (bytes[offset + 4] & 0xFFL) << 24
-        | (bytes[offset + 5] & 0xFFL) << 16
-        | (bytes[offset + 6] & 0xFFL) << 8
-        | (bytes[offset + 7] & 0xFFL);
+  static long longFromByteBuffer(ByteBuffer bytes) {
```

Comment:
Because this is now very simple I would move this directly in the TraceId/SpanId. I think that will avoid on order change in the TraceId case as well a checkArgument.